src/main/scala/com/microsoft/hyperspace/util/MinMaxAnalysisUtil.scala

```diff
@@ -777,4 +778,31 @@ object MinMaxAnalysisUtil extends MinMaxAnalysis {
   def analyze(df: DataFrame, colNames: Seq[String]): String = {
     analyze(df, colNames, "text")
   }
+
+  private def latestIndexEntry(spark: SparkSession, indexName: String): IndexLogEntry = {
+    val idxManager = Hyperspace.getContext(spark).indexCollectionManager
+    val latestVer = idxManager.getIndexVersions(indexName, Seq("ACTIVE")).max
+    val indexEntry = idxManager.getIndex(indexName, latestVer).get
+
+    if (!Seq("CoveringIndex", "ZOrderCoveringIndex").contains(indexEntry.derivedDataset.kind)) {
+      throw HyperspaceException(s"Does not support index type: ${indexEntry.derivedDataset.kind}")
```

Comment:
What if this is checked in the analyzeIndex? And is it possible to check supported index types more systemically?
If someone implement a new type of "CoveringIndex", it might be hard to come here to add it in the supported type.